modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1405,6 +1434,58 @@ void GDScriptAnalyzer::resolve_class_body(GDScriptParser::ClassNode *p_class, co
 		resolve_pending_lambda_bodies();
 	}
 
+	// Resolve base abstract class/method implementation requirements.
+	if (!p_class->is_abstract) {
+		HashSet<StringName> implemented_funcs;
+		int num_unimplemented_funcs = 0;
+		int num_abstract_funcs = 0;
+		String recent_method_signature;
```

Comment:
In my opinion, this is too overcomplicated implementation to display errors. Function signatures are not needed, GDScript does not support overloads, just the name of the first unimplemented method would be enough. Also I'm not sure that counting the number of abstract and/or unimplemented methods is really important, we can just stop after the first error found.

Reply:
The full signature is for better ergonomics. In my opinion, the user would want to know what parameters/return type the unimplemented function uses. They could of course go to the abstract class themselves and look, but this simplifies the workflow.

I'm okay with not counting how many unimplemented functions / how many abstract methods in a non abstract class there are. I still think both errors should show up if a problem exists, so I'll make a solution that allows for both to be emitted at the same time if both problems exist.

Reply:
If you forgot to add abstract in the base class, then there is no point in displaying the signature of same class method, the problem is not in the signature, but in a missing `@abstract`.

If you forgot to implement an inherited method, then you still have to look at the base class. Autocompletion should insert the method signature when overridden, just like with native virtual methods. This is not implemented at the moment, but there is a PR.

In my opinion, there is no need for a signature here, the error just tells you that there is something wrong with the class. And besides, we have no other errors with so many details. Displaying the parent signature when overriding a method makes sense because it explains the cause of the error for a specific incompatible method, and not for the entire class.

Reply:
After thinking about it, you're right on both of those points. Especially if the autocomplete will fill in the signature in the future, that would solve my grievance with not knowing the signature.

I've updated the PR and now only one of the two error messages will display even if both problems are present, and I removed the method signature in favor of just the class::method. I'm going to also revert my change of making a function that returns the method signature, given now that the function would only be called in one place.